simple_network_sim/network_of_populations.py

```diff
@@ -283,18 +284,22 @@ def doInternalProgressionAllNodes(dictOfNodeInternalStates, currentTime, disease
 
 NetworkOfPopulation = namedtuple("NetworkOfPopulation", ["progression", "states", "graph", "infectionMatrix"])
 
-def createNetworkOfPopulation(disasesProgressionFn, populationFn, graphFn):
+def createNetworkOfPopulation(disasesProgressionFn, populationFn, graphFn, ageInfectionMatrixFn):
     with open(disasesProgressionFn) as fp:
         progression = loaders.readCompartmentRatesByAge(fp)
     with open(populationFn) as fp:
         population = loaders.readPopulationAgeStructured(fp)
     graph = loaders.genGraphFromContactFile(graphFn)
 
-    # TODO: read this from a file
-    infectionMatrix = {}
-    for ageA in progression.keys():
-        for ageB in progression.keys():
-            infectionMatrix.setdefault(ageA, {})[ageB] = 0.2
+    infectionMatrix = loaders.MixingMatrix(ageInfectionMatrixFn)
+
+    agesInInfectionMatrix = set(infectionMatrix)
+    for age in infectionMatrix:
+        agesInInfectionMatrix.update(infectionMatrix[age])
```

Comment:
This is just a union of all age ranges from the infection matrix, right? Are we really trying to deal with a scenario where the row and column headers are not the same? If anything, I thought we should have a check to ensure that they are identical (i.e., change the update line to an `assert set(infectionMatrix[age]) == agesInInfectionMatrix`).

Reply:
I just wanted to make sure all the age ranges in all data structures are the same. Do you see any harm in checking all the ocurrences of it rather than relying that it was already checked somewhere else?

Reply:
I agree on checking that all the age ranges are identical across all data structures. I just wasn't sure why you were setting `agesInInfectionMatrix` to be the union of all possible age ranges in the infection matrix. For instance, if the column headers are `[0,5)` and `[17,30)`, and row 1 has the header `[5,17)` and row 2 has the header `[30,40)`, then `agesInInfectionMatrix` would contains `['[0,5)', '[5,17)', '[17,30)', '[30,40)']`.  Instead, I would've expect us to throw an error because the row headers don't line up with the column headers at all.

Reply:
I see your point. I should just assert that every row is the same as the columns here. You're right.